test/coverage/communication/NotificationSinkTest.cpp

```diff
@@ -22,39 +22,67 @@
 namespace {
 using MsgDiff = google::protobuf::util::MessageDifferencer;
 using namespace uprotocol::communication;
+namespace UriValidator = uprotocol::datamodel::validator::uri;
 
 class NotificationSinkTest : public testing::Test {
 protected:
 	// Run once per TEST_F.
 	// Used to set up clean environments per test.
-	void SetUp() override { buildValidSinkURI(); }
+	void SetUp() override;
 	void TearDown() override {}
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
 	NotificationSinkTest() = default;
 	~NotificationSinkTest() = default;
 
-	void buildValidSinkURI(const std::string& authority = "192.168.1.10") {
-		testSink_.set_authority_name(authority);
-		testSink_.set_ue_id(0x18000);
-		testSink_.set_ue_version_major(0x1);
-		testSink_.set_resource_id(0x0);
-	}
+	void buildDefaultSourceURI();
+	void buildValidNotificationURI();
+	void buildInValidNotificationURI();
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 
-	uprotocol::v1::UUri testSink_;
+	uprotocol::v1::UUri testTopicUUri_;
+	uprotocol::v1::UUri testInvalidTopicUUri_;
+	uprotocol::v1::UUri testDefaultSourceUUri_;
 	size_t capture_count_ = 0;
 	uprotocol::v1::UMessage capture_msg_;
 
 public:
 	void handleCallbackMessage(const uprotocol::v1::UMessage& message);
 };
 
+void NotificationSinkTest::SetUp() {
+	buildValidNotificationURI();
+	buildInValidNotificationURI();
+	buildDefaultSourceURI();
+}
+
+void NotificationSinkTest::buildDefaultSourceURI() {
+	testDefaultSourceUUri_.set_authority_name("192.168.1.10");
+	testDefaultSourceUUri_.set_ue_id(0x00011102);
+	testDefaultSourceUUri_.set_ue_version_major(0x1);
+	testDefaultSourceUUri_.set_resource_id(0x0);
```

Comment:
The resource_id should be non-zero for sources.  Probably >= 0x8000 for Notifications.

Reply:
After speaking with Pranav, the zero resource_id needs to stay until this PR (https://github.com/eclipse-uprotocol/up-cpp/pull/200) is merged.  I'm marking this PR as approved so it can merge.